viewing_party/party.py

```diff
@@ -0,0 +1,194 @@
+    in order to return the value
+    """
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+    }
+    for value in new_movie.values():
+        if value == None:
+            return None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    """
+    This function ammends the user_data "watched" key with new movie information
+    """
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """
+    This function ammends the user_data "watchlist" key with new movie information
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    """
+    This function takes a movie title, reviews user_data "watchlist", if movie matches a title in watchlist
+    the function will move movie to "watched" and remove from "watchlist"
+    """
+    watchlist = user_data["watchlist"]
+    watched = user_data["watched"]
+    for movie in watchlist:
+        if movie["title"] == title:
+            watched.append(movie)
+            watchlist.remove(movie)
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    """
+    This function will return the average rating for a "watched" movies in user_data, 
+    will return 0.0 if nothing contained in "watchlist"
+    """
+    sum_rating = 0
+    average_rating = 0.0
+    watched = user_data["watched"]
+    if len(watched) == 0:
+        return average_rating
+    else:
+        for movie in watched:
+            sum_rating += movie["rating"]
+            average_rating = sum_rating / len(watched)
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    """
+    This function will return the most watched genre in user_data "watched" data.
+    """
+    genre_dict = {}
+    if len(user_data["watched"]) == 0:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            if movie["genre"] not in genre_dict:
+                genre_dict[movie["genre"]] = 1
+            else:
+                genre_dict[movie["genre"]] += 1
+        popular_genre = max(genre_dict, key=genre_dict.get)
+        return popular_genre
+    # max_freq = 0
+    # max_genre = None
+    # for genre, freq in genre_dict.items():
+    #     if freq > max_freq:
+    #         max_freq = freq
+    #         max_genre = genre
+    # I commmented this section out because I used max(, key.get) first, but I don't know if I fully understand it
+
+def get_unique_watched(user_data):
+    """
+    Given a user_data "watched", this function will compare to friend's "watched" data and return
+    unique values of user_data
+    """
+    user_watched = user_data["watched"]
+    unique_watched = []
+    for movie_title in user_watched:
+        if movie_title not in merge_friend_watch_lists(user_data):
+            unique_watched.append(movie_title)
+    return(unique_watched)
+
+def merge_friend_watch_lists(user_data):
```

Comment:
👍 this could also be a good place to use a `set`.  

Other than that this is a good helper function.